src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -911,7 +909,33 @@ public override BoundNode VisitConversion(BoundConversion conversion)
             }
         }
 
-        private void GetLambdaId(SyntaxNode syntax, ClosureKind closureKind, int closureOrdinal, out MethodDebugId topLevelMethodId, out int lambdaOrdinal)
+        private DebugId GetTopLevelMethodId()
+        {
+            return slotAllocatorOpt?.MethodId ?? new DebugId(_topLevelMethodOrdinal, CompilationState.ModuleBuilderOpt.CurrentGenerationOrdinal);
+        }
+
+        private DebugId GetClosureId(SyntaxNode syntax, ArrayBuilder<ClosureDebugInfo> closureDebugInfo)
+        {
+            Debug.Assert(syntax != null);
+
+            DebugId closureId;
+            DebugId previousClosureId;
+            if (slotAllocatorOpt != null && slotAllocatorOpt.TryGetPreviousClosure(syntax, out previousClosureId))
+            {
+                closureId = previousClosureId;
+            }
```

Comment:
Minor point: Perhaps remove 'previousClosureId' and pass 'closureId' to TryGetPreviousClosure instead. Then the if part can be removed as well. Same comment for VB.